main.py

```diff
@@ -122,4 +135,7 @@ def main(core_context: det.core.Context, hparams: Dict[str, Any]):
     except KeyError:
         distributed = None
     with det.core.init(distributed=distributed) as core_context:
-        main(core_context, hparams)
+        if core_context.distributed.local_rank == 0:
+            main(core_context, hparams)
+        else:
+            logging.info("Local rank !=0 does nothing.")
```

Comment:
device_map="auto" [won't work on multiple nodes](https://github.com/huggingface/transformers/issues/24747), only on multiple GPUs in one node. However, w/o limiting to one rank only, all ranks start to use all the GPUs, which leads to OOMs.

I will create a ticket to add deepspeed inference with auto tensor parallelism to the eval branch and work on that later on.

Reply:
I think I'd be pretty confused by that `info` message if I saw it as a user, and might think that my non-chief GPUs are not being used at all.

Reply:
you are right, let me remove it.

Reply:
Very very minor thing haha. Just something I noticed